helm/nginx-ingress-controller-app/templates/controller-service-nodeport.yaml

```diff
@@ -6,8 +6,8 @@ kind: Service
 metadata:
   annotations:
     prometheus.io/scheme: http
-    prometheus.io/scrape: "true"
-    prometheus.io/port: "10254"
+    prometheus.io/scrape: 'true'
+    prometheus.io/port: '{{ .Values.controller.metrics.port }}'
```

Comment:
WHy the annotations are in this `Service` and not in the headless one?

Reply:
AFAIK these annotations are "classic" way of publishing metadata about prometheus exporter targets, while the new headless metrics Service is for prometheus-operator way of doing the same thing. 

Classic metadata on NodePort (azure) and LoadBalancer (aws) types of Service doesn't harm prometheus-operator when new metrics Service is enabled.

prometheus-operator (ServiceMonitor) needs just metrics Service spec + Endpoints coming out of metrics Service. From my test it seems that absence of classic annotations didn't seem to do any harm, discovery works. Still, please check/verify if you see something strange in discovered Targets on Prometheus on `giraffe - 5k5hg` TC, it's still up from my tests, you can access it locally after running
```
$ kubectl port-forward -n monitoring prometheus-prometheus-operator-promet-prometheus-0 9090
```

There's also Grafana with nginx dashboard, you can access Grafana locally after running
```
$ kubectl port-forward -n monitoring prometheus-operator-grafana-845fd8b6cb-b9bh2 3000
```
admin password is prom-operator (default from chart)

@JosephSalisbury as our Prometheus expert, can you also please have a look too?

Reply:
From POV all is good. Thx for the explanation